pkg/syncer/cnsoperator/controller/cnsregistervolume/util.go

```diff
@@ -117,6 +117,31 @@ func getK8sStorageClassName(ctx context.Context, k8sClient clientset.Interface,
 	return "", errors.New(msg)
 }
 
+// isVolumeBoundByOtherPVC checks if vsphere block volume in cnsregistervolume is bound to any other PVC
+// returns error if true.
+func isVolumeBoundByOtherPVC(ctx context.Context, k8sClient clientset.Interface, instance *cnsregistervolumev1alpha1.CnsRegisterVolume) error {
```

Comment:
It looks like this method wont detect duplicate registration when its invoked concurrently. Can we fix this? 

On further thinking, should we have a validating webhook admission controller that checks this and rejects duplicate register volume requests?

Reply:
How about we fix the bug as this validation in reconciler for now? I can fix the concurrency issue for the method.

The proposal for having an admission controller for entry level checks makes sense, but can we track it separately as an enhancement?

Reply:
I think we should not do both. Can we find out if handling concurrency in this method will fix the cases? For example, what happens if there are parallel requests and we have 2 instances of CSI pods running? Do we have leader election for CnsRegisterVolume controller? Do we ever end up in a case where we have 2 active CnsRegisterVolume controller running? If yes, then we should not merge this PR.

Reply:
>Do we have leader election for CnsRegisterVolume controller? Do we ever end up in a case where we have 2 active CnsRegisterVolume controller running? If yes, then we should not merge this PR.

This is not possible. There will be only 1 active CnsRegisterVolume controller running. 
See this - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/cmd/syncer/main.go#L98 for reference.

Given this, I don't think concurrency is a problem.

Reply:
Can we test this to be sure? Even if there is one active controller running I'm not sure if its a good idea to "try" to deal with duplicate requests in the controller logic(its too late). If we want to prevent duplicate requests, it should be handled in admission controller.